src/agent/src/rpc.rs

```diff
@@ -1802,7 +1802,10 @@ fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
                 return false;
             }
         };
-        if line.starts_with("SigCgt:") {
+
+        // check SigBlk, SigIgn, SigCgt
+        if line.starts_with("SigBlk:") || line.starts_with("SigIgn:") || line.starts_with("SigCgt:")
+        {
             let mask_vec: Vec<&str> = line.split(':').collect();
             if mask_vec.len() != 2 {
                 warn!(sl!(), "parse the SigCgt field failed");
```

Comment:
It seems that the loop should continue to try the next sigmask

Reply:
Thanks @Tim-Zhang, change to filter to check signal mask.